.github/workflows/portal-loop.yml

```diff
@@ -46,3 +48,57 @@ jobs:
           push: ${{ github.event_name != 'pull_request' }}
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+
+  test-portal-loop-docker-compose:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: "Checkout"
+        uses: actions/checkout@v4
+
+      - name: "Setup the images"
+        run: |
+          cd misc/loop
+
+          docker compose build
+          docker compose pull
+          docker compose up -d
+
+      - name: "Test1 - Portal loop start gnoland"
+        run: |
+          while
+            block_height=$(curl -s localhost:26657/status | jq -r '.result.sync_info.latest_block_height')
+            echo "Current block height: $block_height"
+            [[ "$block_height" -lt 10 ]]
+          do
+            sleep 1
+          done
+
+          curl -s localhost:26657/status | jq
+
+      - name: "Buid new gnolang/gno image"
+        run: |
+          docker build -t ghcr.io/gnolang/gno -f Dockerfile.gnoland.release .
+
+      - name: "Wait for new docker image"
+        run: |
+          ip_addr=$(cat misc/loop/traefik/gno.yml | grep -o "http://.*:26657")
+          while
+            new_ip_addr=$(cat misc/loop/traefik/gno.yml | grep -o "http://.*:26657")
+            echo "${ip_addr} -> ${new_ip_addr}"
+            [[ "${ip_addr}" == ${new_ip_addr} ]]
+          do
+            sleep 5
+          done
+
+      - name: "Test2 - Wait portal-loop start new image"
+        run: |
+          while
+            block_height=$(curl -s localhost:26657/status | jq -r '.result.sync_info.latest_block_height')
+            echo "Current block height: $block_height"
+            [[ "$block_height" -lt 10 ]]
+          do
+            sleep 5
+          done
+          docker ps -a
+          curl -s localhost:26657/status | jq
```

Comment:
why check the status here instead of doing it inside the docker compose file? we can use the `healthcheck` yaml config param to define when a container is considered up and running and add a timeout there too. By doing that we can avoid these `while` shell blocks and just check that the `docker compose up` command finished correctly.

Reply:
I don't just want to check the container is healthy (signing blocks).
i want to be sure we started again from less than block 10, (so from scratch) and not from previous portal loop which will continue to work in case of issue

Reply:
I agree this can be probably be achieved using `HEALTCHECK` and `--wait-timeout` (Maximum duration to wait for the project to be running|healthy). On the other hand I don't think this would improve readability by far.